DirectEmployers/MyJobs
# Logging and profiling

## What's this PR do?
Adds a page profiler for staff users for all the times we're told X.jobs is slow, and need to figure out what's so unique about X.jobs.

Logs cache hits (except for sessions).

Logs solr queries.

## Screenshots (if appropriate)
Profiler:
![prof](https://user-images.githubusercontent.com/2728248/29191981-9e737fd8-7ded-11e7-958f-1ee524ccdeb0.png)

Cache hits:
```
[10/Aug/2017 16:32:39] INFO [cache_backends:22] {"action": "cache hit", "cache_key": ":2017-08-10:www.my.jobs:SeoSite", "site_id": 42669}
[10/Aug/2017 16:32:39] INFO [cache_backends:22] {"action": "cache hit", "cache_key": ":2017-08-10:excluded_view_sources", "site_id": 42669}
[10/Aug/2017 16:32:39] INFO [cache_backends:22] {"action": "cache hit", "cache_key": ":2017-08-10:custom_excluded_view_sources", "site_id": 42669}
[10/Aug/2017 16:32:39] INFO [cache_backends:22] {"action": "cache hit", "cache_key": ":2017-08-10:www.my.jobs:config:2", "site_id": 42669}
[10/Aug/2017 16:32:39] INFO [cache_backends:22] {"action": "cache hit", "cache_key": ":2017-08-10:views.decorators.cache.cache_header.www.my.jobs-5044-2-348.62fa920e664eaba38461d412ed8089f0.en.America/New_York", "site_id": 42669}
[10/Aug/2017 16:32:39] INFO [cache_backends:22] {"action": "cache hit", "cache_key": ":2017-08-10:views.decorators.cache.cache_page.www.my.jobs-5044-2-348.GET.62fa920e664eaba38461d412ed8089f0.d41d8cd98f00b204e9800998ecf8427e.en.America/New_York", "site_id": 42669}
```

Solr queries:
```[10/Aug/2017 16:57:29] INFO [seo.search_backend:51] {"query_string": "full_loc:(Virtual, AND USA)", "duration": 0.030219078063964844, "additional_args": [], "site_id": 1, "additional_kwargs": {"bf": "recip(ms(NOW/HOUR,salted_date),1.8e-9,1,1)", "start_offset": 0, "fields": ["apply_info", "city", "company", "company_canonical_microsite", "company_enhanced", "company_exact", "company_slab", "country", "country_short", "date_new", "date_updated", "django_ct", "django_id", "guid", "highlighted", "id", "is_posted", "link", "location", "location_exact", "reqid", "score", "state", "state_short", "text", "title", "title_exact", "uid", "django_id", "django_ct", "score", "id"], "facets": {"full_loc_exact": {}, "city_slab_exact": {}, "state_slab_exact": {}, "buid": {}, "moc_slab_exact": {}, "company_slab_exact": {}, "country_slab_exact": {}, "title_slab_exact": {}, "mapped_moc_slab_exact": {}}, "facet_sort": "count", "sort_by": "score desc", "multi_src": true, "narrow_queries": ["on_sites:(0)"], "highlight": true, "end_offset": 30, "facet_limit": 30, "facet_mincount": 1}}
[10/Aug/2017 16:57:29] INFO [seo.search_backend:51] {"query_string": "*:*", "duration": 0.011981010437011719, "additional_args": [], "site_id": 1, "additional_kwargs": {"bf": "recip(ms(NOW/HOUR,salted_date),1.8e-9,1,1)", "start_offset": 0, "fields": ["apply_info", "city", "company", "company_canonical_microsite", "company_enhanced", "company_exact", "company_slab", "country", "country_short", "date_new", "date_updated", "django_ct", "django_id", "guid", "highlighted", "id", "is_posted", "link", "location", "location_exact", "reqid", "score", "state", "state_short", "text", "title", "title_exact", "uid", "django_id", "django_ct", "score", "id"], "facets": {"full_loc_exact": {}, "city_slab_exact": {}, "state_slab_exact": {}, "buid": {}, "moc_slab_exact": {}, "company_slab_exact": {}, "country_slab_exact": {}, "title_slab_exact": {}, "mapped_moc_slab_exact": {}}, "facet_sort": "count", "sort_by": "score desc", "multi_src": true, "narrow_queries": ["on_sites:(0 OR 0)"], "end_offset": 1, "facet_limit": 250, "facet_mincount": 1}}
```

## How should this be manually tested?
Profiler:
1. Go to www.my.jobs?prof
2. Observe that it's a normal search results page.
3. Log in
4. Go to www.my.jobs?prof
5. Observe profiling results.

Logging:
1. Add 'console' to the loggers in default_settings.py
2. Update cache to use Memcached (you may have to set up Memcached to make this work) with the custom backend.
3. Load a search results page.
4. Observe all the wonderful logging in your console.

## Any background context you want to provide?
This is mostly to proactively help us answer questions or observe changes in system behavior. If you have any suggestions for how it can be improved, please contribute!
